pkg/netpol/connlist/connlist.go

```diff
@@ -375,6 +376,11 @@ func (ca *ConnlistAnalyzer) getConnectionsList(pe *eval.PolicyEngine) ([]Peer2Pe
 			if !ca.includePairOfWorkloads(srcPeer, dstPeer) {
 				continue
 			}
+			// skip self-looped connection,
+			// i.e. a connection from workload to itself (regardless existence of replicas)
+			if srcPeer.String() == dstPeer.String() {
+				continue
+			}
```

Comment:
maybe this should also be checked within `includePairOfWorkloads()`

Reply:
right , moved it